lib/scheduler.pm

```diff
@@ -116,12 +116,45 @@ sub load_yaml_schedule {
     return 0;
 }
 
-sub _import_test_data_from_yaml {
-    my ($yaml_file) = @_;
-    my $include_yaml = YAML::Tiny::LoadFile(dirname(__FILE__) . '/../' . $yaml_file);
+sub _import_test_data_included {
+    my ($test_data) = shift;
+
+    if (defined(my $import = $test_data->{$include_tag})) {
+        # Allow both lists and scalar value for import "!include" key
+        if (ref $import eq 'ARRAY') {
+            for my $include (@{$import}) {
+                _import_test_data_from_yaml($include);
+            }
+        }
+        else {
+            _import_test_data_from_yaml($import);
+        }
+        delete $test_data->{$include_tag};
+    }
+}
+
+sub _ensure_include_not_present {
+    my ($include_yaml) = shift;
     if (exists $include_yaml->{$include_tag}) {
-        die "Error: test_data can only be defined in a dedicated file for data\n";
+        die "Error: please define in the file only the content of your test_data," .
+          " without including tag $include_tag\n";
     }
+}
+
+sub _import_test_data_from_yaml {
+    my ($path, $allow_included) = @_;
```

Comment:
As we now have 2 variables, we should using hash:
```
my ($args) = @_;
print $args->{path};
```
And call method by:
```perl
_import_test_data_from_yaml( path => $path, allow_included => 1 );
```

Reply:
Perl and signatures :) ... CI failed, it is not a hash ref what is passed to the function in that way. I did it with a hash, let's see CI likes it.

Reply:
Right, I missed curly brackets, I've fixed an example above.
Feel free to change $args to %args to access hash directly and not by reference. Then you can access keys like `$args{path}`.